lib/ui/page/login/widget/prefix_button.dart

```diff
@@ -0,0 +1,108 @@
+// option) any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License v3.0 for
+// more details.
+//
+// You should have received a copy of the GNU Affero General Public License v3.0
+// along with this program. If not, see
+// <https://www.gnu.org/licenses/agpl-3.0.html>.
+
+import 'package:flutter/material.dart';
+
+import '/themes.dart';
+
+/// Stylized button with the optional [prefix] widget.
+class PrefixButton extends StatelessWidget {
+  const PrefixButton({
+    super.key,
+    this.title = '',
+    this.onPressed,
+    this.style,
+    this.prefix,
+  });
+
+  /// Title of this [PrefixButton].
+  final String title;
+
+  /// [TextStyle] of the [title].
+  final TextStyle? style;
+
+  /// Callback, called when this button is pressed.
+  final void Function()? onPressed;
+
+  /// [Widget] to display as a prefix.
+  final Widget? prefix;
+
+  @override
+  Widget build(BuildContext context) {
+    final style = Theme.of(context).style;
+
+    final BorderRadius borderRadius = BorderRadius.circular(
+      15 * 0.72,
+    );
+    return Stack(alignment: Alignment.centerLeft, children: [
+      Container(
+        constraints: const BoxConstraints(
+          maxWidth: double.infinity,
+        ),
+        alignment: Alignment.center,
+        decoration: BoxDecoration(
+          color: style.colors.onPrimary,
+          borderRadius: borderRadius,
+          border: Border.all(
+            width: 0.5,
+            color: style.colors.secondary,
+          ),
+        ),
+        child: Material(
+          color: style.colors.transparent,
+          borderRadius: borderRadius,
+          child: InkWell(
+            borderRadius: borderRadius,
+            onTap: onPressed,
+            hoverColor: style.colors.onBackgroundOpacity7,
+            child: Container(
+              constraints: const BoxConstraints(
+                minHeight: 46,
+                maxHeight: double.infinity,
+              ),
+              padding: const EdgeInsets.symmetric(
+                horizontal: 8 * 0.7,
+                vertical: 6 * 0.7,
+              ),
+              child: Row(children: [
+                const SizedBox(width: 8),
+                Expanded(
+                    child: DefaultTextStyle.merge(
+                  maxLines: 2,
+                  overflow: TextOverflow.ellipsis,
+                  textAlign: TextAlign.center,
+                  style: style.fonts.titleLarge.copyWith(
+                    color: onPressed == null
+                        ? style.colors.onBackgroundOpacity40
+                        : style.colors.onBackground,
+                  ),
+                  child: Center(
+                    child: Text(
+                      title,
+                    ),
+                  ),
+                )),
```

Comment:
А тут добавить, чтобы форматирование было лучше:

```suggestion
                ),),
```

В прошлом ревью я неоднократно Вам указывал на такие места. Пожалуйста, не пропускайте их и будьте внимательнее. Замечания пишутся не для того, чтобы поправить пару мест и забыть.